src/backend/distributed/commands/vacuum.c

```diff
@@ -73,13 +76,24 @@ static CitusVacuumParams VacuumStmtParams(VacuumStmt *vacstmt);
 void
 PostprocessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
 {
-	int relationIndex = 0;
-	List *vacuumRelationList = ExtractVacuumTargetRels(vacuumStmt);
-	List *relationIdList = NIL;
 	CitusVacuumParams vacuumParams = VacuumStmtParams(vacuumStmt);
 	LOCKMODE lockMode = (vacuumParams.options & VACOPT_FULL) ? AccessExclusiveLock :
 						ShareUpdateExclusiveLock;
-	int executedVacuumCount = 0;
+
+
+	if (vacuumParams.options & VACOPT_VACUUM)
+	{
+		/*
+		 * We commit the current transaction here so that the global lock
+		 * taken from the shell table for VACUUM is released, which would block execution
+		 * of shard placements.
+		 */
+		CommitTransactionCommand();
```

Comment:
is there any downside to doing this for completely unrelated vacuums? 

should we err on the side of caution and only do this if there are actual Citus tables?

Reply:
so should we error for

```sql
vacuum citus_table, postgres_table
```

Reply:
I think mixing postgres and Citus tables should be ok, but Citus should do as little as possible when no Citus tables are involved.

Reply:
I tried some commands concurrently with certain breakpoints, couldn't see an issue but to be on the safe side we now return early if there are no citus tables in the vacuum. However if there are mixed workload, we won't return early and continue.